account_asset_management/models/account_move.py

```diff
@@ -122,11 +117,15 @@ def button_draft(self):
     def _reverse_move_vals(self, default_values, cancel=True):
         move_vals = super()._reverse_move_vals(default_values, cancel)
         if move_vals["type"] not in ("out_invoice", "out_refund"):
+            assets_to_unlink_ids = []
             for line_command in move_vals.get("line_ids", []):
                 line_vals = line_command[2]  # (0, 0, {...})
-                asset = self.env["account.asset"].browse(line_vals["asset_id"])
-                asset.unlink()
-                line_vals.update(asset_profile_id=False, asset_id=False)
+                asset_id = line_vals.get("asset_id")
```

Comment:
AFAIK, generated depreciation move lines are also linked through this same field, so you should only try to remove assets when they are linked to the first one (the invoice one). Or are you forbidding the reversal of this through the raise about the asset being linked to depreciations?